### PR TITLE
Document package architecture

### LINKED ISSUE #655: Document API structure and architecture guidelines
https://github.com/woodpecker-ci/woodpecker/pull/651#issuecomment-1004045621

### PR DESCRIPTION
**Pull request by 6543**
to make it easyer for devs to find the right place for code

close #655

### CODE REVIEW

File: docs/docs/92-development/05-architecture.md [hunk 1]
```diff
@@ -6,4 +6,41 @@
...
+| `cli/**`   | code only cli tool does need                                 | `pipeline`, `shared`, `woodpecker-go`
+| `server/**`| code only server will need                                   | `pipeline`, `shared`
+| `shared/**`| code shared for all three main tools (go help utils)         | only std and external libs
+| `woodpecker-go/**` | go client for server rest api                        | std
+
+### Server
+
+| package            | meaning                                         | import
+|--------------------|-------------------------------------------------|----------
+| `server/api/**`    | handle web requests from `server/router`        | `pipeline`, `server/(badges\|ccmenue\|logging\|model\|pubsub\|queue\|remote\|shared\|store)`, `shared`, (TODO: mv `server/router/middleware/session`)
```
Comment (6543) [inline docs/docs/92-development/05-architecture.md:24]: as we can see we should move more stuff out of it

either create a new package if we can "discover" a new scope or distribute it to other packages ...

Comment (6543) [reply docs/docs/92-development/05-architecture.md:24]: cc @anbraten

**Comment by 6543** [comment]
that's just stupid - we need to escape - but escape char will get rendered :/

**Comment by 6543** [comment]
I'll just write down the current state (and what it should be) with TODOs in source ... we can refactor this later

File: docs/docs/92-development/05-architecture.md [hunk 2]
```diff
@@ -6,4 +6,44 @@
...
+| `server/ccmenu/**`  | generate xml ccmenu for pipelines               | `../model`
+| `server/grpc/**`    | gRPC server agents can connect to               | `pipeline/rpc/**`, `../logging`, `../model`, `../pubsub`, `../queue`, `../remote`, `../pipeline`, `../store`
+| `server/logging/**` | logging lib for gPRC server to stream logs while running | std
+| `server/model/**`   | structs for store (db) and api (json)           | std
+| `server/plugins/**` | plugins for server                              | `../model`, `../remote`
+| `server/pipeline/**`| orchistrate pipelines                          | `pipeline`, `../model`, `../pubsub`, `../queue`, `../remote`, `../store`, `../plugins`
+| `server/pubsub/**`  | pubsub lib for server to push changes to the WebUI | std
+| `server/queue/**`   | queue lib for server where agents pull new pipelines from via gRPC | `server/model`
+| `server/remote/**`  | remote lib for server to connect and handle forge specific stuff | `shared`, `server/model`
+| `server/router/**`  | handle REST API (and all middleware) and serve route `web` | `shared`, `../api`, `../model`, `../remote`, `../store`, `../web`
```
Comment (anbraten) [inline docs/docs/92-development/05-architecture.md:35]:
```suggestion
| `server/router/**`  | handle REST API (and all middleware) and serve UI and WebUI config | `shared`, `../api`, `../model`, `../remote`, `../store`, `../web`
```

**Review by anbraten — APPROVED** [review]

**Pull request merged**